store/src/java/com/zimbra/cs/service/mail/ItemAction.java

```diff
@@ -540,24 +557,42 @@ protected StringBuilder proxyRemoteItems(Element action, Map<String, StringBuild
             action.addAttribute(MailConstants.A_FOLDER, iidFolder.toString());
         }
 
-        StringBuilder successes = new StringBuilder();
+        String opStr = getOperation(request);
+
+        ItemActionResult result = null;
+        if (opStr.equals(MailConstants.OP_HARD_DELETE))
+        {
+            result = new DeleteActionResult();
+        }
+        else
+        {
+            result = new ItemActionResult();
+        }
         for (Map.Entry<String, StringBuilder> entry : remote.entrySet()) {
             // update the <action> element to reference the subset of target items belonging to this user...
             String itemIds = entry.getValue().toString();
             action.addAttribute(MailConstants.A_ID, itemIds);
             // ... proxy to the target items' owner's server...
             String accountId = entry.getKey();
+            // TODO: Add 'advanced' option to trigger Create / Delete additional information response
+
             Element response = proxyRequest(request, context, accountId);
             // ... and try to extract the list of items affected by the operation
             try {
                 String completed = response.getElement(MailConstants.E_ACTION).getAttribute(MailConstants.A_ID);
-                successes.append(completed.length() > 0 && successes.length() > 0 ? "," : "").append(completed);
+                List<String> ids = new ArrayList<String>();
+                for (String id: completed.split(","))
+                {
+                	ids.add(id);
+                }
+                result.appendSuccessIds(ids);
```

Comment:
Good idea!
I'll add another method to do that.  I'll leave the existing one for bulk updates.